packages/std/src/never.rs

```diff
@@ -0,0 +1,21 @@
+/// Never can never be instantiated. This can be used in places
+/// where we want to ensure that no error is returned, such as
+/// the `ibc_packet_receive` entry point.
+///
+/// Once the ! type is stable, this is not needed anymore.
+/// See <https://github.com/rust-lang/rust/issues/35121>.
+pub enum Never {}
+
+impl core::fmt::Debug for Never {
+    fn fmt(&self, _f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        // This is unreachable because no instance of Never can exist
+        unreachable!()
+    }
+}
+
+impl core::fmt::Display for Never {
```

Comment:
Maybe add a comment that this is needed to satisfy the requirement of error type in cosmwasm results